src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,281 @@
+#include "src/PluginStructs/P145_data_struct.h"  // MQ-xxx sensor specific data
+
+#define PLUGIN_145
+#define PLUGIN_ID_145     145           // plugin id
+#define PLUGIN_NAME_145   "Gases - MQxxx (MQ135 CO2, MQ3 Alcohol) [TESTING]" // "Plugin Name" is what will be dislpayed in the selection list
+#define PLUGIN_VALUENAME1_145 "level"   // variable output of the plugin. The label is in quotation marks
+#define PLUGIN_145_DEBUG  false         // set to true for extra log info in the debug
+
+// Static, per plugin instance, data stored in the generic ESPeasy structures see usage defined above
+// PCONFIG(n)    : stores an integer (8)
+// PCONFIG_FLOAT : stors a float (4)
+// PCONFIG_LONG  : stores a long (4, shared with PCONFIG_ULONG)
+// PCONFIG_ULONG : stores an unsigned long (4, shared with PCONFIG_LONG)
+#define P145_PCONFIG_RLOAD       PCONFIG_FLOAT(0)
+#define P145_PCONFIG_RZERO       PCONFIG_FLOAT(1)
+#define P145_PCONFIG_REF         PCONFIG_FLOAT(2)
+#define P145_PCONFIG_FLAGS       PCONFIG(0)
+#define P145_PCONFIG_TEMP_TASK   PCONFIG(1)
+#define P145_PCONFIG_TEMP_VAL    PCONFIG(2)
+#define P145_PCONFIG_HUM_TASK    PCONFIG(3)
+#define P145_PCONFIG_HUM_VAL     PCONFIG(4)
+#define P145_PCONFIG_SENSORT     PCONFIG(5)
+
+// PIN/port configuration is stored in the following:
+#define CONFIG_PIN_AIN      CONFIG_PIN1
+
+// A plugin has to implement the following function
+// ------------------------------------------------
+boolean Plugin_145(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number = PLUGIN_ID_145;
+      Device[deviceCount].Type = DEVICE_TYPE_ANALOG;
+      Device[deviceCount].VType = Sensor_VType::SENSOR_TYPE_SINGLE;
+      Device[deviceCount].Ports = 0;
+      Device[deviceCount].PullUpOption = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 1;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+    
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_145);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_145));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // FormSelector with all predefined "Sensor - Gas" options
+      String options[P145_MAXTYPES] = {};
+      int optionValues[P145_MAXTYPES] = {};
+      int x = P145_data_struct::getNbrOfTypes();
+      if (x > P145_MAXTYPES) 
+      {
+        x = P145_MAXTYPES;    // Clip to prevent array boundary out of range access
+      }
+      for (int i=0; i<x; i++)
+      {
+        options[i] = concat(P145_data_struct::getTypeName(i), F(" - ")) + (String)P145_data_struct::getGasName(i);
```

Comment:
Do not use C-style casts like this. Especially not for type `String`
A C-style cast is a bit like a `reinterpret_cast` which essentially tells the compiler "interpret the data starting at that address as if it were initialized/constructed as the type I tell you it is".
However for a `String` class, this just isn't correct.
The `String` class does have some buffer memory and some flags on how to interpret this buffer. It either is upto 12 bytes of a C-style char array, or a pointer. But if you try to use this `String` object, you have to hope the functions trying to do something with its data isn't concluding this text data may need to be intepreted as a pointer since the bits telling so are undefined.

Simple fix here is to change `(String)P145_data_struct::getGasName(i)` into `String(P145_data_struct::getGasName(i))`.